src/main/resources/lib/nextjs-proxy/proxy.es6

```diff
@@ -39,14 +40,87 @@ const errorResponse = function (url, status, message, req, renderSingleComponent
     }
 };
 
+function cookiesArrayToObject(array) {
+    const cookies = {};
+    if (array?.length > 0) {
+        array.forEach(cookie => {
+            const indexEq = cookie.indexOf("=");
+            if (indexEq > 0) {
+                const indexSc = cookie.indexOf(";");
+                cookies[cookie.substring(0, indexEq)] = cookie.substring(indexEq + 1, indexSc > 0 ? indexSc : undefined);
+            }
+        });
+    }
+    return cookies;
+}
+
 // lib-http response is different from the one controller awaits
 function okResponse(libHttpResponse) {
     return {
         body: libHttpResponse.body || libHttpResponse.bodyStream,
         status: libHttpResponse.status,
         contentType: libHttpResponse.contentType,
         headers: libHttpResponse.headers,
+        cookies: cookiesArrayToObject(libHttpResponse.cookies),
+    }
+}
+
+function doRequest(req, renderSingleComponent) {
+    const frontendUrl = req.url;
+    const xpSiteUrl = req.headers['xpBaseUrl'];
+
+    const response = httpClientLib.request(req);
+
+    if (Math.floor(response.status / 100) === 3) {
```

Comment:
`String(response.status).startsWith('3')`?

Reply:
Humanitarian vs mathematician ways 😆

Reply:
`response.status >= 300 && response.status < 400` is probably the easiest to understand though